core/templates/pages/exploration-editor-page/services/exploration-automatic-text-to-speech.service.spec.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the ExplorationAutomaticTextToSpeechService.
+ */
+
+// TODO(#7222): Remove the following block of unnnecessary imports once
+// the code corresponding to the spec is upgraded to Angular 8.
+import { UpgradedServices } from 'services/UpgradedServices';
+// ^^^ This block is to be removed.
+
+require('pages/exploration-editor-page/' +
+  'services/exploration-automatic-text-to-speech.service.ts');
+
+describe('Exploration Automatic Text To Speech Service', function() {
+  let eattss = null;
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    var ugs = new UpgradedServices();
+    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
+      $provide.value(key, value);
+    }
+  }));
+
+  beforeEach(angular.mock.inject(function($injector) {
+    eattss = $injector.get('ExplorationAutomaticTextToSpeechService');
+  }));
+
+  it('should check the datatype', function() {
+    expect(eattss._isValid(true)).toBe(true);
+    expect(eattss._isValid(false)).toBe(true);
+  });
+
+  it('should enable and disable Automatic Text-to-speech', function() {
+    eattss.savedMemento = true;
+    expect(eattss.isAutomaticTextToSpeechEnabled()).toBe(true);
+    eattss.savedMemento = false;
+    expect(eattss.isAutomaticTextToSpeechEnabled()).toBe(false);
+  });
+
+  it('should call the saveDisplayedValue function', function() {
```

Comment:
Provide a better description for this spec. Actually, I think you should do more assertions than just checking if an internal function is called. We expect to test the output of a function and not its implementation (maybe in the future the `toggleAutomaticTextToSpeech` is implemented using another approach and this test will need to refactored.